ospf6d/ospf6_asbr.c

```diff
@@ -2316,6 +2316,37 @@ void ospf6_asbr_redistribute_reset(struct ospf6 *ospf6)
 	}
 }
 
+void ospf6_asbr_redistribute_reconfigure(struct ospf6 *ospf6)
+{
+	int type;
+	struct ospf6_redist *red;
+	char *rmap_name = NULL;
+
+	for (type = 0; type <= ZEBRA_ROUTE_MAX; type++) {
+		if (type == ZEBRA_ROUTE_OSPF6)
+			continue;
+		red = ospf6_redist_lookup(ospf6, type, 0);
+		if (!red)
+			continue;
+
+		if (type == DEFAULT_ROUTE) {
+			ospf6_redistribute_default_set(
+				ospf6, ospf6->default_originate);
+			continue;
+		}
+		if (ROUTEMAP_NAME(red)) {
+			rmap_name = malloc(sizeof(red));
+			rmap_name = strdup(ROUTEMAP_NAME(red));
+		}
+		ospf6_asbr_redistribute_unset(ospf6, red, type);
+		if (rmap_name) {
+			ospf6_asbr_routemap_set(red, rmap_name);
+			free(rmap_name);
```

Comment:
There was no need of a malloc/strdup() and then another free(). All these system calls should be avoided as much as possible.